### PR TITLE
Update @unlock-protocol/unlock-js to 0.3.1 in unlock-app

### PR DESCRIPTION
**Pull request by cnasc**
# Description

The second of a 2-part dep upgrade, this including code changes to meet the new API

Edit: Absorbing the 0.3.1 upgrade as well

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by cnasc** [comment]
Forgot to cherry-pick a commit, I'll grab it in the morning and the test will go green

**Comment by julien51** [comment]
Thanks for this @cnasc !

File: unlock-app/src/middlewares/providerMiddleware.ts [hunk 1]
```diff
@@ -47,13 +47,9 @@ export async function initializeUnlockProvider(
   unlockProvider: any,
   dispatch: any
 ) {
-  const { key, emailAddress, password } = action
   try {
-    await unlockProvider.connect({ key, password })
+    await unlockProvider.connect(action)
```
Comment (julien51) [inline unlock-app/src/middlewares/providerMiddleware.ts]: Could you remind me of `action` is for? Maybe put it as a comment on JSdoc?

Comment (cnasc) [reply unlock-app/src/middlewares/providerMiddleware.ts]: Yup, I'll add in that it's the `GOT_ENCRYPTED_PRIVATE_KEY` action. (there's actually just such a comment on the other side, in the provider, but it makes sense to also have it here.)

**Pull request merged**